Echo enable/disable in configuration file.

patrykkrz opened the pull request:
ESP_ECHO is defined but not used. 
In normal mode (no debug), echo is unnecessary.
With a large amount of data sent, data back to uart receiver.

MaJerle commented:
This code needs mofication before I can merge it, because:
- Echo is enabled by default and to disable it, command ATE0 must be sent automatically.
- Instead of deleted code, code for enable echo should be in #IFDEF part.

patrykkrz commented:
What do You mean "command ATE0 must be sent automatically" ?
I send the command ATE in ESP_Init (I get parameter from #define ESP_ECHO),
after AT command.

As far as I understand you:
- ATE0 send always, when user not define ESP_ECHO or define ESP_ECHO is 0.
- If user define ESP_ECHO, send ATE1

Is that what you mean?



2017-02-13 22:51 GMT+01:00 Tilen Majerle <notifications@github.com>:

> This code needs mofication before I can merge it, because:
>
>    - Echo is enabled by default and to disable it, command ATE0 must be
>    sent automatically.
>    - Instead of deleted code, code for enable echo should be in #IFDEF
>    part.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MaJerle/ESP8266_AT_Commands_parser/pull/39#issuecomment-279534461>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AYj1GI8WB7QIv2v0FNtBwH7_n9uGpYmLks5rcNB6gaJpZM4L_xDQ>
> .
>

MaJerle commented:
ATE code must be in library. You just deleted it. If echo is disabled, ate0 should be send on init. Of enabled, ate1 should be sent.

patrykkrz commented:
Echo is always on after power up ESP module. So I think I have to send the command ATE0 if the echo off and don't send ATE command, when echo is on.  Would you agree?

Should I to leave ECHO_ON in the configuration file (esp8266_config_templete.h) or definition ECHO_ON must be only in esp8266.h ?

MaJerle commented:
You deleted code to enable echo in protothread call. I dont know why you did that but you did. If echo is disabled by conf, then ATE0 must be send to module. If it is enabled, ATE1 should be sent. But bo katter what, you should not delete any code.

patrykkrz commented:
I added code in protothread (PT_Thread_BASIC), not removed. My commit
changed 3 files (+35 -2). This is a misunderstanding?

2017-02-14 18:53 GMT+01:00 Tilen Majerle <notifications@github.com>:

> You deleted code to enable echo in protothread call. I dont know why you
> did that but you did. If echo is disabled by conf, then ATE0 must be send
> to module. If it is enabled, ATE1 should be sent. But bo katter what, you
> should not delete any code.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MaJerle/ESP8266_AT_Commands_parser/pull/39#issuecomment-279783167>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AYj1GKAbcGIj8gyV5MuELaTIRxvLplSuks5rceo0gaJpZM4L_xDQ>
> .
>

MaJerle commented:
It was misunderstanding, yes.

patrykkrz commented:
I fixed a comment. 
Is now everything is ok?
